src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int MULTIPLIER = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(Objects.hash(key), key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentBucket = table[index];
+            while (currentBucket != null) {
+                if (Objects.equals(currentBucket.key, key)) {
+                    currentBucket.value = value;
+                    return;
+                }
+                if (currentBucket.next == null) {
+                    currentBucket.next = newNode;
+                    break;
+                }
+                currentBucket = currentBucket.next;
+            }
+        }
+        size++;
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        size = 0;
+        threshold *= MULTIPLIER;
+        int newCapacity = table.length * MULTIPLIER;
+        Node<K, V>[] oldTable = table;
+        table = new Node[newCapacity];
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
```

Comment:
```suggestion
        private Node(int hash, K key, V value, Node<K, V> next) {
```
Why do we need a public? Let's make it private.